corehq/apps/users/views/__init__.py

```diff
@@ -724,12 +725,22 @@ def __call__(self, request, uuid, **kwargs):
             logout(request)
             return HttpResponseRedirect(request.path)
 
-        invitation = SQLInvitation.objects.filter(uuid=uuid).first()
+        # Recently-sent invitations will use a URL based on UUID
+        try:
+            invitation = SQLInvitation.objects.get(uuid=uuid)
+        except (SQLInvitation.DoesNotExist, ValidationError):
+            invitation = None
+
+        # Older invitations, created before PR#26975, will use a URL based on SQL id
         if not invitation:
             try:
                 invitation = SQLInvitation.objects.get(id=int(uuid))
-            except (ValueError, SQLInvitation.DoesNotExist):
-                invitation = SQLInvitation.objects.filter(couch_id=uuid).first()
+            except (SQLInvitation.DoesNotExist, ValueError):
+                invitation = None
+
+        # The oldest invitations, created before PR#26686, will use a URL based on couch id.
+        if not invitation:
+            invitation = SQLInvitation.objects.filter(couch_id=uuid).first()
```

Comment:
non blocking/can be taken as a followup after this is resolved:
why not just filter by all three at once and use whatever is returned?

Reply:
I hadn't thought of that but am going to leave this as-is since this is temporary code that'll be removed in https://github.com/dimagi/commcare-hq/pull/26687